proposals/text/7462-rag-evaluation.md

```diff
@@ -0,0 +1,243 @@
+- Title: Proposal for presentation of RAG evaluation results
+- Decision driver: David S. Batista
+- Start Date: 2024-04-03
+- Proposal PR: #7462
+- Github Issue or Discussion: https://github.com/deepset-ai/haystack/issues/7398
+
+# Summary
+
+Add a new component to Haystack allowing users interact with the results of evaluating the performance of a RAG model.
+
+
+# Motivation
+
+RAG models are one of them most popular use cases for Haystack. We are adding support for evaluations metrics, but there is no way to present the results of the evaluation.
+
+
+# Detailed design
+
+The output results of an evaluation pipeline composed of `evaluator` components are passed to a `RAGPipelineEvaluation`
+(this is a placeholder name) which stores them internally and acts as an interface to access and present the results.
+
+Example of the data structure that the `RAGPipelineEvaluation` class will receive for initialization:
+
+```python
+
+data = {
+    "queries": {
```

Comment:
I wouldn't use this format but let's not discuss that as part of this proposal. I would use keys for the inputs that the metrics also receive, questions (and/or query id), contexts, responses.

Reply:
I did not understand your suggestion - what you suggest (as far as I've understood it) is already there but inside `queries` (now changed to `inputs`) or do you mean something else?